src/SecretsService/Providers/EnvironmentReader.cs

```diff
@@ -1,29 +1,26 @@
 using System;
 using System.IO;
-using System.Collections.Generic;
-using System.Threading.Tasks;
 using ACMTTU.NoteSharing.Shared.DataContracts;
-using ACMTTU.NoteSharing.SecretsService.Models;
-using Newtonsoft.Json;
+using ACMTTU.NoteSharing.Shared.SDK.Clients;
 
 namespace ACMTTU.NoteSharing.SecretsService.Providers
 {
     public static class EnvironmentReader
     {
-        public static EnvironmentVariablePayload GetEnvironmentVariableFromKeyVault(EnvironmentVariableOptions option)
+        public static EnvironmentVariablePayload GetEnvironmentVariableFromKeyVault(ClientOptions option)
         {
             EnvironmentVariablePayload payload = new EnvironmentVariablePayload();
 
             switch (option)
```

Comment:
Does the service return the credentials for ALL stages to the client?

Also the exception that's thrown when the `default:` case is hit should be more like $"Invalid resource requested: {option}"

Edit: Looks like it's filtered out in https://github.com/ACMTTU/acm-note-sharing-platform/pull/14/commits/900ac8eacdf7d13866b46abdef6bca7e626272ac#diff-ead3c0ace27db71ecda8b29deae742c4R19 👌

Reply:
yes, I thought it would be easier to return all 3 environment variables and have the SDK

Reply:
And yeah that default case should never get hit. That's why I say something really weird happend

Reply:
Yeah, that sounds good to me then. Do clients need to decode the base64 secret when using the SDK then? Just saw your Postman screenshots.

Reply:
All the services that want to use a database just need to do this...

```csharp
   DatabaseClientFactory dbClientFactory = new DatabaseClientFactory()
   CosmosClient client = await dbClientFactory.GetClientAsync()
```

But the more i think about it... I should really be using dependency injection

```csharp
class Application {
  private ClientFactory clientFactory;

  public Application(IClientFactory clientFactory) {
    this.clientFactory = clientFactory;
  }

  public async dynamic FindAllTheDataInDatabase() {
    using (CosmosClient databaseClient = await this.clientFactory.GetClientAsync<ClientOptions.Database>()) {
      // Do whatever needs to get done with the database
    }
    
  }
}
```

Reply:
Both work so I'll let you decide on what you'd like everyone else to do. Other than that, LGTM 👍

Reply:
DI is the way to go if you want to mock your DB to test your DB logic.

Reply:
Yes I did that :)